packages/sanity/src/router/RouterProvider.tsx

```diff
@@ -114,17 +144,55 @@ export function RouterProvider(props: RouterProviderProps): ReactElement {
     [onNavigate, resolveIntentLink],
   )
 
+  const [routerState, stickyParams] = useMemo(() => {
+    if (!state._searchParams) {
+      return [state, null]
+    }
+    const {_searchParams, ...rest} = state
+    const [sticky, restParams] = partition(_searchParams, ([key]) => STICKY.includes(key))
+    if (sticky.length === 0) {
+      return [state, null]
+    }
+    return [{...rest, _searchParams: restParams}, sticky]
+  }, [state])
+
   const router: RouterContextValue = useMemo(
     () => ({
       navigate,
       navigateIntent,
+      navigateStickyParam: handleNavigateStickyParam,
       navigateUrl: onNavigate,
       resolveIntentLink,
       resolvePathFromState,
-      state,
+      state: routerState,
+      stickyParams: Object.fromEntries(stickyParams || []),
     }),
-    [navigate, navigateIntent, onNavigate, resolveIntentLink, resolvePathFromState, state],
+    [
+      handleNavigateStickyParam,
+      navigate,
+      navigateIntent,
+      onNavigate,
+      resolveIntentLink,
+      resolvePathFromState,
+      routerState,
+      stickyParams,
+    ],
   )
 
   return <RouterContext.Provider value={router}>{props.children}</RouterContext.Provider>
 }
+const STICKY: string[] = []
```

Comment:
Currently there are no sticky params defined, however in content releases, the `perspective` key will be added here (the first current use of sticky params), but they can be used in other instances also